src/jit/hwintrinsicxarch.cpp

```diff
@@ -337,11 +462,54 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic        intrinsic,
     {
         return gtNewIconNode(issupported);
     }
-    else if (!issupported)
+    // - calling to unsupported intrinsics must throw PlatforNotSupportedException
+    // - 64-bit (r64) intrinsics must throw PlatforNotSupportedException on 32-bit platform
+    else if (!issupported || is64bitOnlyIntrinsicOn32bitPlatform(category, callType))
```

Comment:
nit: It doesn't seem worthwhile to force the `is64bitOnlyIntrinsicOn32bitPlatform` check here, since it will only be hit by a very small number of the intrinsics.

Reply:
But it unifies the thow code. And the checking is very simple, I think the branch prediction can easily sovle it.

Reply:
I'm still not convinced (although I don't feel particularly strongly about fixing it either since it only impacts 32-bit).

My thinking was that we have ~1250 intrinsics total and ~50 of those that take or return an int64 (and therefore could require a 64-bit register). And since we are already checking the category below (and I think we will need that long term), it might have been worth moving it.